salt/_modules/kubernetesmod.py

```diff
@@ -2276,3 +2276,32 @@ def node_set_taints(node_name, taints, **kwargs):
             raise CommandExecutionError(exc)
     finally:
         _cleanup(**cfg)
+
+
+def node_drain(node_name, **kwargs):
+    '''
+    Drain the the node identified by the name `node_name`.
+
+    CLI Examples::
+
+        salt '*' kubernetes.node_drain node_name="minikube"
+    '''
+
+    '''
+    Data:
+        * List of namespaces
+        * List of pods to evict (for a given namespace)
+        * List of evicted pods to delete (for a given namespace)
+
+    Get list of all namespaces (need to handle continue parameter?)
+    For given node:
```

Comment:
shouldn't we mark the node as unschedul-able (cordon) so no new pods enter while we are draining ?

Reply:
cordon should have been made before calling drain...

Reply:
Yes (as does `kubectl drain`), otherwise the draining process will never converge.

Reply:
yep...